gems/aws-sigv4/sig/signer.rbs

```diff
@@ -0,0 +1,70 @@
+module Aws
+  module Sigv4
+    class Signer
+      def initialize: (
+        service: String,
+        region: String,
+        access_key_id: String,
+        secret_access_key: String,
+        session_token?: String?,
+        unsigned_headers?: Array[String]?,
+        uri_escape_path?: bool,
+        apply_checksum_header?: bool,
+        signing_algorithm?: Symbol,
```

Comment:
Since this is a limited set of valid symbols (sigv4, sigv4a, sigv4-s3express), we could limit that here.  Not sure if we should or not though. Thoughts?

Reply:
I thought it would be better to restrict it to the valid symbols only, so I’ve updated the code accordingly.
2a780e6cc1c843a179acdebeb5382297ca390aba